### PR TITLE
Support for traces in sessions

### PR DESCRIPTION
**Pull request by alexandrosfilios**

### CODE REVIEW

File: platform/view/view/context.go [hunk 1]
```diff
@@ -72,6 +76,12 @@ type MutableContext interface {
 
 // Context gives a view information about the environment in which it is in execution
 type Context interface {
+	// StartSpanFrom creates a new child span from the passed context
+	StartSpanFrom(ctx context.Context, name string, opts ...trace.SpanStartOption) (context.Context, trace.Span)
```
Comment (adecaro) [inline platform/view/view/context.go:80]: can we move these two functions to a different interface that is imported by Context?

File: platform/view/view/session.go [hunk 2]
```diff
@@ -23,6 +24,7 @@ type Message struct {
 	FromPKID     []byte // PK identifier of the caller
 	Status       int32  // Message Status (OK, ERROR)
 	Payload      []byte // Payload
+	Ctx          context.Context
```
Comment (adecaro) [inline platform/view/view/session.go:27]: is this guaranteed to be serializable?

File: platform/view/view/session.go [hunk 3]
```diff
@@ -53,9 +55,15 @@ type Session interface {
 	// Send sends the payload to the endpoint
 	Send(payload []byte) error
 
+	// SendWithContext sends the payload to the endpoint with the passed context
```
Comment (adecaro) [inline platform/view/view/session.go:58]: does this mean that the context is part of the message or the context is used to control the send operation itself?

File: platform/view/view/session.go (hunk 2 above)
Comment (alexandrosfilios) [reply platform/view/view/session.go:27]: it isn't, but it doesn't have to be

File: platform/view/view/session.go (hunk 3 above)
Comment (alexandrosfilios) [reply platform/view/view/session.go:58]: no it's part of the message. it carries some information. in this case the span context

Comment (adecaro) [reply platform/view/view/session.go:58]: I would clarify in the comment then

Comment (alexandrosfilios) [reply platform/view/view/session.go:58]: No but the context doesn't have to do only with the span. The span uses the context but the context has more functions in general, e.g. canceling the context, or adding baggage values

**Review by adecaro — APPROVED** [review]
LGTM

**Pull request merged**